lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js

```diff
@@ -363,6 +488,21 @@ describe('LMSFilePicker', () => {
         fileList.prop('noFilesMessage').props.href,
         'https://fake_help_link'
       );
+      // After first file fetch, we're at the top level, so the no-files message
+      // should operate in its "course" context
+      assert.isFalse(fileList.prop('noFilesMessage').props.inSubfolder);
+    });
+
+    it('renders no-file message in folder context if in a subfolder', async () => {
+      const wrapper = renderFilePicker({ withBreadcrumbs: true });
+      await waitForElement(wrapper, 'FileList');
+
+      act(() => changePath(wrapper, fakeFolders[0]));
+      const fileList = await waitForElement(wrapper, 'FileList');
```

Comment:
I think you want to wait for not just the file list but for the file list to be in a non-loading state, since the file list was already visible prior to this call.